linchpin/__init__.py

```diff
@@ -122,6 +122,8 @@ def runcli(ctx, config, workspace, verbose, version, creds_path):
 
     if workspace is not None:
         ctx.workspace = os.path.realpath(os.path.expanduser(workspace))
+    else:
```

Comment:
This would revert back the PR I sent last week, #313. Before that the PR #320 has already broken because it removes the self.workspace from the LinchpinContext class, once the first default setting for this variable. :(